### PR TITLE
Added more parameters for plugin: number of workers, cpu limit, memory limit

### PR DESCRIPTION
**Pull request by Chris210634**
@rudolphpienaar @jbernal0019 @danmcp,

Added optional parameters for plug-ins.
These are optional. If they are not specified, the default will be used during plug-in registration.

### CODE REVIEW

File: chrisapp/base.py [hunk 1]
```diff
@@ -206,6 +209,9 @@ def get_json_representation(self):
         repres['selfpath'] = self.SELFPATH
         repres['selfexec'] = self.SELFEXEC
         repres['execshell'] = self.EXECSHELL
+        repres['number_of_wokers'] = self.NUMBER_OF_WORKERS
```
Comment (danmcp) [inline chrisapp/base.py]: This doesn't seem to account for min and max values.

**Comment by danmcp** [comment]
LGTM

@rudolphpienaar @jbernal0019 Please review and merge.

**Comment by jbernal0019** [comment]
Hey @danmcp @Chris210634 @rudolphpienaar I guess that "These are optional. If they are not specified, the default will be used during plug-in registration." means that they don't need to be enforced to be defined in the plugin app code. Right?  If they are not  defined in the plugin app code then during plugin registration the defaults will be added to the plugin representation,  right?

So I also guess that those parameters don't necessary need to be shown in the store. Right?

**Comment by jbernal0019** [comment]
Or should we enforce those class variables to be defined in the plugin app code? Even the the user just leave them untouched like this:

    MAX_NUMBER_OF_WORKERS = 1  # Override with integer value
    MIN_NUMBER_OF_WORKERS = 1  # Override with integer value
    MAX_CPU_LIMIT         = '' # Override with millicore value as string, e.g. '2000m'
    MIN_CPU_LIMIT         = '' # Override with millicore value as string, e.g. '2000m'
    MAX_MEMORY_LIMIT      = '' # Override with string, e.g. '1Gi', '2000Mi'
    MIN_MEMORY_LIMIT      = '' # Override with string, e.g. '1Gi', '2000Mi'

**Comment by jbernal0019** [comment]
Oh, I realize that they are in the superclass anyway. So if they are optional they don't need to be enforced and they will be in the representation anyway with those default values in my previous comment

**Comment by jbernal0019** [comment]
Yes @danmcp But we will need to make sure that the default values used at plugin registration are the same as here:

`MAX_NUMBER_OF_WORKERS = 1  # Override with integer value
MIN_NUMBER_OF_WORKERS = 1  # Override with integer value
MAX_CPU_LIMIT         = '' # Override with millicore value as string, e.g. '2000m'
MIN_CPU_LIMIT         = '' # Override with millicore value as string, e.g. '2000m'
MAX_MEMORY_LIMIT      = '' # Override with string, e.g. '1Gi', '2000Mi'
MIN_MEMORY_LIMIT      = '' # Override with string, e.g. '1Gi', '2000Mi'`

**Comment by danmcp** [comment]
I would rather not specify defaults in chrisapp (other than #workers).  The default should come from the compute cluster or CUBE unless the plugin writer specifies a value.  The reason being is those defaults will be difficult to change once embedded into the plugin image.

**Pull request merged**